python/hail/tests/tests.py

```diff
@@ -1098,6 +1098,19 @@ def test_union(self):
         self.assertTrue(r1.same(vds))
 
     def test_trio_matrix(self):
+        """
+        This test depends on certain properties of the TDT VCF
```

Comment:
I think we should decouple this from TDT. Can you make a separate ``trio_matrix.fam`` with just the two lines you're actually using?

Then you can simplify this explanation as:
```
This test depends on the simple structure of ``trio_matrix.fam``, namely that the trios are complete and each sample appears exactly once.  For example, with quads, the trio_matrix method will duplicate the parents appropriately, but the genotypes_table and samples_table orthogonal paths would require another duplication/explode that we haven't written.
```

How hard-nosed should I be able testing this in some form on a quad / incomplete trios? I suppose if mendel errors is reimplemented in terms of this and still passes, that would be convincing:
```
Fam0 Indiv0 0 0 1 0
Fam1 Son1 Dad1 Mom1 1 1
Fam1 Dtr1 Dad1 Mom1 2 2
Fam1 Dad1 0 0 1 0
Fam1 Mom1 0 0 2 0
Fam2 Son2 Dad2 Mom2 1 2
Fam2 Dad2 0 0 1 2
Fam2 Mom2 0 0 2 1
Fam3 Son3 Dad3 0 2 0
Fam3 Dad3 0 0 2 2
Fam4 Indiv4 0 0 1 1
```

Reply:
The integration tests of implementing TDT / de novo / mendel errors with this should test it just fine!